enhancements/network/ocp-on-smart-nics.md

```diff
@@ -0,0 +1,313 @@
+OpenShift operators cannot be deployed on a node with a Smart NIC, and
+so all masters and at least a few workers will need to have "dumb"
+NICs.
+
+(While the exact architectural constraints may change in the future,
+it is likely that some customers will always want the possibility of
+only having Smart NICs on a subset of nodes.)
+
+### Overall Cluster Architecture
+
+There are several possible overall architectures:
+
+  - The Smart NICs could be nodes within the same OCP cluster as their
+    hosts. We would need to use some combination of labels and taints
+    to distinguish "real" nodes from Smart NIC nodes. (Among other
+    things, the Smart NICs would be their own `MachineConfigPool`,
+    separate from the masters and workers.) This may require changes
+    to tolerations and node selectors on existing DaemonSets, both
+    within OCP and in third-party operators or customer workloads (and
+    it may turn out that this approach is unworkable for this reason).
+
+  - The Smart NICs could be part of a second cluster running in
+    parallel with the primary cluster, with the control plane of the
+    NIC cluster running as pods within the primary cluster.
+
+  - The Smart NICs could be part of a second cluster which is
+    completely separate from the primary cluster, with its own master
+    nodes. (In some external discussion of this approach, it was
+    assumed that the masters in this scenario would have to be on ARM
+    hosts, but this isn't required; you can have a mix of x86_64
+    master nodes and ARM Smart NIC "worker" nodes.)
+
+  - The Smart NICs could be part of a second cluster, where the
+    control planes for both clusters are running in a different
+    cluster, ROKS-style.
+
+The latter two approaches allow for more isolation between the hosts
+and the NICs, since potentially even the cluster administrators of the
+primary cluster might not be able to influence the NIC cluster. This
+would be especially important if additional primary-cluster security
+and monitoring functions were offloaded to the NICs in the future.
+
+### Initial Development With Beta BlueField-2 Hardware
+
+[NOTE: We are confirming the timing/availability of various hardware.
+It may turn out that it doesn't make sense for us to worry about these
+cards.]
+
+Although we will eventually want to use IPMI to control and provision
+the NICs, just like we do for the hosts with bare-metal IPI, this will
+not work with the pre-GA BlueField-2 cards that we currently have
+access to. Instead, we will have to communicate with the NICs over PCI
+from the host. The most future-compatible approach is probably to
+create an IPMI proxy that can run on each host, claiming its own
+separate IP address on the provisioning network, and then proxying
+commands to the NIC via the PCI "RShim" interface. (Thus, we will have
+to first install OCP on the node, then deploy a (highly-privileged)
+DaemonSet to the node to run the IPMI proxy, then install OCP on the
```

Comment:
When provisioning the smart NIC the host will lose cluster network connection. It will be hard to debug if the IPMI proxy pod doesn't work as expected.